cmd/run_test.go

```diff
@@ -212,3 +213,83 @@ func TestRunScriptErrorsAndAbort(t *testing.T) {
 		})
 	}
 }
+
+// alarmist is a mock module that do a panic
+type alarmist struct {
+	vu modules.VU
+}
+
+var _ modules.Module = &alarmist{}
+
+func (a *alarmist) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &alarmist{
+		vu: vu,
+	}
+}
+
+func (a *alarmist) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]interface{}{
+			"panic": a.panic,
+		},
+	}
+}
+
+func (a *alarmist) panic(s string) {
+	panic(s)
+}
+
+func TestRunScriptPanicsErrorsAndAbort(t *testing.T) {
+	t.Parallel()
+
+	modules.Register("k6/x/alarmist", new(alarmist))
```

Comment:
:question: to a reviwers, is there a way to do `Unregister`, should we implement something like that? :thinking: 

e.g. use case:
```golang
defer func (){
  modules.Register("k6/x/alarmist")
}()
```

Reply:
The only use case is tests which make this less interesting. In #2399 I am circumventing this by having a test in a package. Or at least have the tests for a given module in a separate package. So that the module is only registered for those tests. 
(FYI package tests are compiled separately from each other so that is the only place that a `_test.go` file will be executed)

There was some [discussion](https://github.com/grafana/k6/issues/1858#issuecomment-787818358) long time ago about having non global registries and so on, but we ended up with the current situation. 

I guess we can work on the `js/runner` to maybe be able to give it a list of modules :shrug: but that likely will look even worse.

so :shrug:

Reply:
>(FYI package tests are compiled separately from each other so that is the only place that a _test.go file will be executed)

That's what I know, but even inside the package, it's changing of the global state, which I'd like to clean up after, but for sure, it's not a big deal.

But in general, is your suggestion to move this into a separate package? :thinking:

Reply:
> That's what I know, but even inside the package, it's changing of the global state, which I'd like to clean up after, but for sure, it's not a big deal. 

Yeah and my whole idea in that package is that this tests will be alone in a package - so will not affect anything. And then at the end the process testing just ends, so there is nothing to cleanup :tada: .

> But in general, is your suggestion to move this into a separate package? thinking 

Yes and more *importanly* have the registering in a `init` function so it's called only once. This way in case somebody (I) runs the test with `-count 2` they will not get a failure on the second call to `Register` as it will be called only once ;)

Reply:
Another way instead of `init()` would be to write a small helper that first calls `modules.GetJSModules()` and only calls `modules.Register()` if the module wasn't previously registered. Not sure which is simpler :man_shrugging:

Reply:
I moved the module registration to the `init` since it's hard to move the test to the separated package without  https://github.com/grafana/k6/issues/2459. I'm suggesting to merge this as is, and I'll create a separated PR with moving in the next cycle